controller/authController.js

```diff
@@ -0,0 +1,142 @@
+const User = require("../model/userOtpModel");
+const Otp = require("../model/otpModel");
+const { validationResult } = require("express-validator");
+const nodemailer = require("nodemailer");
+const dotenv = require("dotenv");
+dotenv.config();
+
+exports.signUp = async (req, res, next) => {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      res.status(422).json({ errors: errors.array() });
+      return;
+    }
+    const { username, email } = req.body;
+    const newUser = new User({
+      username,
+      email,
+    });
+    newUser.save().then((result) => {
+      res.send(result);
+    });
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({
+      success: false,
+      message:
+        "We are having some error while completing your request. Please try again after some time.",
+      error: error,
+    });
+  }
+};
+
+exports.loginOtpSend = async (req, res) => {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      res.status(422).json({ errors: errors.array() });
+      return;
+    }
+    const { email } = req.body;
+    const user = await User.findOne({ email });
+    if (!user) {
+      return res.status(400).json({ error: "user not found" });
+    }
+    const login = new Otp({

```

Comment:
why variable name login here?
and why you saving document inside otp collection without otp?

Reply:
removed saving and changed